### PR TITLE
Vof momentum monitoring

### PR DESCRIPTION
**Pull request by blaisb**
# Description of the problem

- For some cases we are running, it is relevant to monitor both the mass and the momentum in VOF simulations

# Description of the solution

- Add momentum monitoring to the mass monitoring. I don't think it's worth it to make multiple files, instead I appended the information to the existing file.

# How Has This Been Tested?

- Tested on some of our regular benchmark and the results make sense.

# Documentation

- Updated documentation accordingly

# Future changes

- The parameter monitor mass conservation is maybe not the most appropriate right now, but I don't want to break the prm file in this PR. We could have a parameter called monitor conservation instead, of maybe monitoring mass and momentum, but i 'd rather postpone this change because it's only esthetics

### CODE REVIEW

File: source/solvers/vof.cc [hunk 1]
```diff
@@ -717,10 +824,18 @@ VolumeOfFluid<dim>::postprocess(bool first_iteration)
...
+            calculate_momentum(this->present_solution,
+                               *multiphysics->get_solution(
+                                 PhysicsID::fluid_dynamics),
+                               fluid_indicators[i]);
+
           if (this_mpi_process == 0)
             {
               std::string fluid_id("");
 
+
```
Comment (hepap) [inline source/solvers/vof.cc:838]:
```suggestion
```

File: applications_tests/lethe-fluid/gls_vof_1_isothermal_compressible_fluid.output [hunk 2]
```diff
@@ -7,8 +7,8 @@ Running on 1 MPI rank(s)...
 ----------------------
 VOF Mass Conservation
 ----------------------
-   time    surface_fluid_0 mass_per_length_fluid_0 surface_fluid_1 mass_per_length_fluid_1 sharpening_threshold 
-0.0000e+00      6.5778e-01              7.8933e-01      3.4222e-01              3.4222e+02           5.0000e-01 
+   time    surface_fluid_0 mass_per_length_fluid_0 momentum-x_fluid_0 momentum-y_fluid_0 surface_fluid_1 mass_per_length_fluid_1 momentum-x_fluid_1 momentum-y_fluid_1 sharpening_threshold
```
Comment (hepap) [inline applications_tests/lethe-fluid/gls_vof_1_isothermal_compressible_fluid.output:10]: Could it be useful to have to have the total mass and total momentum output as well?

**Review by hepap — APPROVED** [review]
Very nice to have this new feature!

File: applications_tests/lethe-fluid/gls_vof_1_isothermal_compressible_fluid.output (hunk 2 above)
Comment (blaisb) [reply applications_tests/lethe-fluid/gls_vof_1_isothermal_compressible_fluid.output:10]: Someone can easily calculate those by adding the two columns in their own script. I'm not sure this is necessary :)

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,12 @@
 All notable changes to the Lethe project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+## [Master] - 2024-03-19
+
+### Added
+
+- MINOR The mass conservation feature of the VOF model now also monitors the momentum of both phases. This should be useful for some cases and to improve the sharpening mechanism. [#1073](https://github.com/lethe-cfd/lethe/pull/1073)
```
Comment (AmishgaAlphonius) [inline CHANGELOG.md:10]:
```suggestion
- MINOR The mass conservation feature of the VOF model now also monitors the momentum of both phases. This should be useful for some cases and improve the sharpening mechanism. [#1073](https://github.com/lethe-cfd/lethe/pull/1073)
```

File: include/solvers/vof.h [hunk 4]
```diff
@@ -205,6 +205,23 @@ class VolumeOfFluid : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+  /**
+   * @brief Calculates the momentum for a given fluid phase.
+   * Used for conservation monitoring.
+   *
+   * @param solution VOF solution (phase fraction)
+   *
+   * @param current_solution_fd current solution for the fluid dynamics
+   *
+   * @param monitored_fluid Fluid indicator (fluid0 or fluid1) corresponding to
+   * the phase of interest.
```
Comment (AmishgaAlphonius) [inline include/solvers/vof.h:217]:
```suggestion
   * @param[in] solution VOF solution (phase fraction)
   *
   * @param[in] current_solution_fd current solution for the fluid dynamics
   *
   * @param[in] monitored_fluid Fluid indicator (fluid0 or fluid1) corresponding to
   * the phase of interest.
```

File: include/solvers/vof.h [hunk 5]
```diff
@@ -205,6 +205,23 @@ class VolumeOfFluid : public AuxiliaryPhysics<dim, GlobalVectorType>
...
+   */
```
Comment (AmishgaAlphonius) [inline include/solvers/vof.h:218]: missing **return**

File: source/core/parameters.cc [hunk 6]
```diff
@@ -1821,7 +1821,7 @@ namespace Parameters
         "calculate mass conservation",
         "true",
         Patterns::Bool(),
-        "Enable calculation of the mass of both fluids in VOF simualtions.");
+        "Enable calculation of the mass of both fluids in VOF simulations.");
```
Comment (AmishgaAlphonius) [inline source/core/parameters.cc:1824]:
```suggestion
        "Enable calculation of the mass and momentum of both fluids in VOF simulations.");
```

File: source/solvers/vof.cc [hunk 7]
```diff
@@ -753,15 +872,32 @@ VolumeOfFluid<dim>::postprocess(bool first_iteration)
...
+              // Add "momentum" columns
+              for (unsigned int d = 0; d < dim; ++d)
+                {
+                  this->table_monitoring_vof.add_value(momentum_names[d],
+                                                       momentum[d]);
+                  this->table_monitoring_vof.set_scientific(momentum_names[d],
+                                                            true);
+                }
+
+              this->table_monitoring_vof.set_scientific(mass_column_name, true);
```
Comment (AmishgaAlphonius) [inline source/solvers/vof.cc:884]: This could be before the momentum to regroup all mass related lines together

**Review by AmishgaAlphonius — APPROVED** [review]
Great addition! Thanks! :)

File: source/solvers/vof.cc [hunk 8]
```diff
@@ -753,15 +872,32 @@ VolumeOfFluid<dim>::postprocess(bool first_iteration)
...
+
```
Comment (AmishgaAlphonius) [inline source/solvers/vof.cc:885]:
```suggestion
```

**Pull request merged**